### ISSUE TITLE
Layer control should have `role="button"`

### ISSUE THREAD

**Issue by Malvoz** [Leaflet/Leaflet#7498]
## Motivation

<img width="200" src="https://user-images.githubusercontent.com/26493779/110512939-fcbd0e00-8105-11eb-9bdd-48cbbab5250d.png" alt>

The layer control is intended as (and behaves similarly to) a button, not a link, and should be conveyed as such to screen reader users.

## Proposed solution

Set attribute [`role="button"`](https://www.w3.org/TR/wai-aria-1.2/#button) on the `<a>` element:

https://github.com/Leaflet/Leaflet/blob/b52961aa469dc17b3710ef7a9c0f13b6f4dba550/src/control/Control.Layers.js#L201-L203

## Additional context

- MDN Documentation: [`role="button"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role)
- Related PR: https://github.com/Leaflet/Leaflet/pull/6869/ (https://github.com/Leaflet/Leaflet/pull/6869/commits/71bc3781f4725bd8d21dd015806b8d1a299ba6ff)

**Comment by johnd0e** [comment]
> The layer control is intended [...]

What about Zoom control?

**Comment by wiesesascha** [comment]
@Malvoz, how does this affect the usage of leaflet?

We've got some strange behavior on some ubuntu/chromium machines.
Click events are not propagated at e.g. the layer controll.
Maybe there's a relation to this issue..

**Comment by Malvoz** [comment]
@wiesesascha that's probably a separate issue. ARIA semantics only affects users of [assistive technology](https://www.w3.org/TR/wai-aria-1.2/#at_support) (ATs), in this case `role="button"` would _convey_ to these users that this element is a button (that it is actionable), and because ATs can intercept keyboard input, in some cases it is even required for an otherwise clickable element to be operable (for example [markers aren't keyboard operable when using a screen reader](https://github.com/Leaflet/Leaflet/issues/7428), presumably since they're not standard interactive elements).

**Comment by vanillajonathan** [comment]
Maybe using a `button` element ought to be considered instead of a `a` element?

**Comment by Malvoz** [comment]
@vanillajonathan Indeed, that'd be optimal (noted in https://github.com/Leaflet/Leaflet/issues/3210#issue-4). It requires a CSS button reset (not very hard to accomplish) but OTOH it's not backwards compatible with people's code, so that's a consideration.
